internal/cri/server/status.go

```diff
@@ -94,5 +95,39 @@ func (c *criService) Status(ctx context.Context, r *runtime.StatusRequest) (*run
 		}
 		resp.Info["lastCNILoadStatus"] = defaultStatus
 	}
+
+	intro, err := c.client.IntrospectionService().Server(ctx, &ptypes.Empty{})
+	if err != nil {
+		return resp, err
+	}
+	cond := &runtime.RuntimeCondition{
+		Type:   ContainerdHasNoDeprecationWarning,
+		Status: true,
+	}
+	if len(intro.Deprecations) > 0 {
+		cond.Status = false
+		reasons := make([]string, len(intro.Deprecations))
+		messages := make(map[string]string, len(intro.Deprecations))
+		for i, d := range intro.Deprecations {
+			reasons[i] = d.ID
+			messages[d.ID] = d.Message
+		}
+		reasonJ, err := json.Marshal(reasons)
+		if err != nil {
+			return resp, err
+		}
+		messageJ, err := json.Marshal(messages)
+		if err != nil {
+			return resp, err
+		}
+		cond.Reason = string(reasonJ)
+		cond.Message = string(messageJ)
+	}
+	resp.Status.Conditions = append(resp.Status.Conditions, cond)
+
 	return resp, nil
 }
+
+// ContainerdHasNoDeprecationWarning is a string for [runtime.RuntimeCondition.Type].
+// Set to true if there is not deprecation warning.
+const ContainerdHasNoDeprecationWarning = "ContainerdHasNoDeprecationWarning"
```

Comment:
```suggestion
const ContainerdHasNoDeprecationWarnings = "ContainerdHasNoDeprecationWarnings"
```

Is plural better here?

Reply:
Fixed